### DISCUSSION TITLE
basePath in custom-server-express only works in dev

### DISCUSSION THREAD

**Discussion by bernardop** [vercel/next.js#13112] [Help] [answered]
Hi all,

I'm having an issue where `Link`s in a custom `express` server with the experimental `basePath` are only working in `dev`. When doing a production build, the `basePath` is not being pre-pended to the pages.

Here's a repo with a minimal setup to reproduce the issue.
https://github.com/bernardop/next-basepath-repro

Any help will be greatly appreciated!

**Comment by timneutkens** [comment, marked as answer]
basePath is a build-time setting not runtime so you have to put it in next.config.js.

**Reply by bernardop** [reply]
Yep! Thanks @timneutkens, after digging in the code I saw that `basePath` looks at `process.env.__NEXT_ROUTER_BASEPATH`, which is set at build time.
